src/Http/Cookie/Cookie.php

```diff
@@ -0,0 +1,387 @@
+<?php
+/**
+ * CakePHP(tm) : Rapid Development Framework (http://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ * @link          http://cakephp.org CakePHP(tm) Project
+ * @since         3.5.0
+ * @license       http://www.opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Cake\Http\Cookie;
+
+use Cake\Utility\Hash;
+use Cake\Utility\Security;
+use DateTimeInterface;
+use InvalidArgumentException;
+use RuntimeException;
+
+class Cookie implements CookieInterface
+{
+
+    use CookieCryptTrait;
+
+    /**
+     * Cookie name
+     *
+     * @var string
+     */
+    protected $name;
+
+    /**
+     * Raw Cookie value
+     */
+    protected $value;
+
+    /**
+     * Cookie data
+     *
+     * If the raw cookie data was a serialized array and was expanded
+     * this property will keep the data.
+     *
+     * @var array
+     */
+    protected $data;
+
+    /**
+     * @var bool
+     */
+    protected $isExpanded = false;
+
+    /**
+     * HTTP only
+     *
+     * @var bool
+     */
+    protected $expiresAt = 0;
+
+    /**
+     * Path
```

Comment:
We could be as cheap as including links to Wikipedia, such as https://en.wikipedia.org/wiki/HTTP_cookie#Domain_and_Path
😼 

Alternatively, we could link to the official RFC for HTTP Cookie:
https://tools.ietf.org/html/rfc6265#section-5.1.4

Personally, I like to (sometimes) augment my code with such links. If I or one of my collegues is not sure about a certain specificiation or the general implementation of something, it's nice to have a link near-by linking directly to a related topic.